src/App.tsx

```diff
@@ -157,239 +286,541 @@ function App() {
+                            {address}
+                          </option>
+                        );
+                      })}
+                    </select>
+
+                    <select
+                      value={selectedMethods[scope] ?? ''}
+                      onChange={(evt) => handleMethodSelect(evt, scope)}
+                    >
+                      <option value="">Select a method</option>
+                      {details.methods.map((method: string) => (
+                        <option key={method} value={method}>
+                          {method}
+                        </option>
+                      ))}
+                    </select>
+
+                    <details className="collapsible-section">
+                      <summary>Invoke Method Request</summary>
+                      <div className="collapsible-content">
+                        <textarea
+                          value={invokeMethodRequests[scope] ?? ''}
+                          onChange={(evt) =>
+                            setInvokeMethodRequests((prev) => ({
+                              ...prev,
+                              [scope]: evt.target.value,
+                            }))
+                          }
+                          rows={5}
+                          cols={50}
+                        />
+                      </div>
+                    </details>
+
+                    <button
+                      id={`invoke-method-${scope}-btn`}
+                      onClick={async () => {
+                        if (selectedMethods[scope]) {
+                          await handleInvokeMethod(
+                            scope,
+                            selectedMethods[scope] ?? '',
+                          );
+                        }
+                      }}
+                    >
+                      Invoke Method
+                    </button>
+
+                    {Object.keys(invokeMethodResults?.[scope] ?? {}).length >
+                      0 &&
+                      Object.entries(invokeMethodResults[scope] ?? {}).map(
+                        ([method, results]) => {
+                          return results.map((result, index) => {
+                            const { text, truncated } = truncateJSON(
+                              result,
+                              150,
+                            );
+
+                            return truncated ? (
+                              <details
+                                key={`${method}-${index}`}
+                                className="collapsible-section"
+                              >
+                                <summary>
+                                  <span className="result-method">
+                                    {method}:
+                                  </span>
+                                  <span className="result-preview">
+                                    {JSON.stringify(result).slice(0, 100)}...
```

Comment:
should this use the truncateJSON helper?

Reply:
cleaned up here: https://github.com/MetaMask/test-dapp-multichain/pull/15/commits/37b80aad7b01e6a0fdcb4f649d3dac278f2c9334